File: poo/python/Triangulo/Triangulo.py
#!/usr/bin/env python
#-*- coding: utf-8 -*-

"""
@author Nathan Cruz <nathan.cruz@unifesp.br>
@created 27/03/14
"""
from __future__ import unicode_literals
from __future__ import print_function

class Triangulo:
    # Atributos #
    __base = None
    __altura = None

    # Construtor #
    def __init__(self, base=10, altura=10):
        self.__base = float(base)
        self.__altura = float(altura)

    # Getters #
    def getBase(self):
        return self.__base
    def getAltura(self):
        return self.__altura

    # Setters #
    def setBase(self, base):
        self.__base = base
    def setAltura(self, altura):
        self.__altura = altura

    # Calcula a area #
    def Area(self):
        return (self.getBase()*self.getAltura())/2.0

    # Imprime dados sobre o objeto #
    def Mostra(self):
        print('\tBase: %.2f' % (self.getBase()))
        print('\tAltura: %.2f' % (self.getAltura()))
        print('\tArea: %.2f' % (self.Area()))


if __name__ == '__main__':

    t1 = Triangulo()
    t2 = Triangulo(3, 6)

    print('Dados do objeto t1:')
    t1.Mostra()
    print('\nDados do objeto t2:')
    t2.Mostra()

    t1.setBase(2)
    t1.setAltura(4)
    t2.setBase(1)
    t2.setAltura(2)

    print('\nDados do objeto t1:')
    t1.Mostra()
    print('\nDados do objeto t2:')
    t2.Mostra()

File: poo/c++/Triangulo/Triangulo.cpp
#include <iostream>

class Triangulo{

    /* Atributos */
    private:
        int base, altura;

    public:

        /* Construtor */
        Triangulo(int b, int a){
            base = b;
            altura = a;
        }

        Triangulo(){
            base = 10;
            altura = 10;
        }

        /* Getters */
        int getBase(){
            return base;
        }

        int getAltura(){
            return altura;
        }

        /* Setters */
        void setBase(int b){
            base = b;
        }

        void setAltura(int a){
            altura = a;
        }

        /* Funcoes auxiliares */
        float Area(){
            return (base*altura)/2;
        }

        void Mostra(){
            std::cout << "\tBase: " << base << "\n";
            std::cout << "\tAltura: " << altura << "\n";
            std::cout << "\tArea: " << this->Area() << "\n";
        }
};

int main(){

    Triangulo t1;
    Triangulo t2 (3, 6);

    std::cout << "Dados do objeto t1:\n";
    t1.Mostra();
    std::cout << "\nDados do objeto t2:\n";
    t2.Mostra();

    t1.setBase(2);
    t1.setAltura(4);
    t2.setBase(1);
    t2.setAltura(2);

    std::cout << "\nDados do objeto t1:\n";
    t1.Mostra();
    std::cout << "\nDados do objeto t2:\n";
    t2.Mostra();


    return 0;
}
